### PR TITLE
Adds 24bit support to op/jack

### PR DESCRIPTION
**Pull request by nefthy**

### CODE REVIEW

File: op/jack.c [hunk 1]
```diff
@@ -91,6 +91,14 @@ static jack_default_audio_sample_t read_sample_le16(const char *buffer)
 	return (jack_default_audio_sample_t)s / (jack_default_audio_sample_t)upper_bound;
 }
 
+static jack_default_audio_sample_t read_sample_le24(const char *buffer)
+{
+	uint32_t u = read_le24(buffer);
+	int32_t s =  u & 0x800000 ? u | 0xFF000000 : u;
```
Comment (mahkoh) [inline op/jack.c]: I don't like it. Better use something like `read_le24i`.

File: op/jack.c [hunk 2]
```diff
@@ -403,6 +418,10 @@ static int op_jack_open(sample_format_t sf, const channel_position_t *cm)
 	if (bits == 16) {
 		sample_bytes = 2;
 		read_sample = sf_get_signed(sf) ? &read_sample_le16 : &read_sample_le16u;
+	} else if (bits == 24) {
+		d_print("signed: %d", sf_get_signed(sf));
```
Comment (mahkoh) [inline op/jack.c]: That doesn't belong here.

File: utils.h [hunk 3]
```diff
@@ -245,6 +245,14 @@ static inline uint32_t read_le32(const char *buf)
 	return b[0] | (b[1] << 8) | (b[2] << 16) | (b[3] << 24);
 }
 
+static inline uint32_t read_le24(const char *buf)
+{
+	const unsigned char *b = (const unsigned char *)buf;
+
+	uint32_t a = b[0] | (b[1] << 8) | (b[2] << 16);
+	return a;
```
Comment (mahkoh) [inline utils.h]: The temporary should be removed.

File: utils.h [hunk 4]
```diff
@@ -245,6 +245,21 @@ static inline uint32_t read_le32(const char *buf)
...
+	const unsigned char *b = (const unsigned char *)buf;
+
+	return b[0] | (b[1] << 8) | (b[2] << 16);
+}
+
+static inline int32_t read_le24i(const char *buf)
+{
+	const unsigned char *b = (const unsigned char *)buf;
+
+	uint32_t a = b[0] | (b[1] << 8) | (b[2] << 16);
```
Comment (mahkoh) [inline utils.h]: This can use read_le24.

**Comment by nefthy** [comment]
@mahkoh I addressed everything, thanks for reviewing.

**Pull request merged**